src/com/clover/remote/client/device/DefaultCloverDevice.ts

```diff
@@ -104,78 +104,78 @@ export abstract class DefaultCloverDevice extends CloverDevice implements Clover
                 case sdk.remotemessage.Method.BREAK:
                     break;
                 case sdk.remotemessage.Method.CASHBACK_SELECTED:
-                    this.notifyObserversCashbackSelected(sdkMessage);
+                    this.notifyObserversCashbackSelected(<sdk.remotemessage.CashbackSelectedMessage> sdkMessage);
```

Comment:
We shouldn't have to cast any of these as the inheritance should take care of all typing.  Are we just trying to be explicit, or is there another reason?

Reply:
Unfortunately they are required.  I had issues generating the typings file with a module due to the nested namespaces.  As a result, I am using namespaces with interfaces which and this pattern does not support inheritance.

Reply:
These casts are required.  CashbackSelectedMessage is a Message, but Message is not a CashbackSelectedMessage.

Reply:
Actually, I believe that sdkMessage is, in this case, a CashbackSelectedMessage (that's what is returned from the parser) which is a Message (as you mentioned) and since the method is looking for a CashbackSelectedMessage, it should be all good.  Did you find that to not be the case?

Reply:
sdk message is a Message (the parser returns a Message) -  var sdkMessage: sdk.remotemessage.Message = this.messageParser.parseMessageFromRemoteMessageObj(rMessage) (~line 99).  A Message is not a CashbackSelectedMessage - the cast is required.

Reply:
👍